ospd/misc.py

```diff
@@ -408,3 +408,31 @@ def go_to_background():
     except OSError as errmsg:
         LOGGER.error('Fork failed: %s', errmsg)
         sys.exit(1)
+
+def create_pid(pidfile):
+    """ Check if there is an already running daemon and creates the pid file.
+    Otherwise gives an error. """
+
+    pid = str(os.getpid())
+
+    if os.path.isfile(pidfile):
+        LOGGER.error("There is an already running process.")
+        return False
+
+    try:
+        f = open(pidfile, 'w')
+        f.write(pid)
+        f.close()
+    except (FileNotFoundError, PermissionError) as e:
```

Comment:
Either use the with statement (safest solution) `ẁith open(...) as f:` or add a finally statement to always close the file. Currently a different exception could possibly occur and the file would not be closed at all.

Reply:
It uses now the with statement.